agenta-web/src/components/EvaluationTable/AICritiqueEvaluationTable.tsx

```diff
@@ -235,35 +244,54 @@ Answer ONLY with one of the given grading or evaluation options.
     }
 
     const runEvaluation = async (rowIndex: number) => {
-        const inputParamsDict = rows[rowIndex].inputs.reduce((acc: {[key: string]: any}, item) => {
-            acc[item.input_name] = item.input_value
-            return acc
-        }, {})
-
-        const columnsDataNames = ["columnData0"]
-        let idx = 0
-        for (const columnName of columnsDataNames) {
-            setRowValue(rowIndex, "evaluationFlow", EvaluationFlow.COMPARISON_RUN_STARTED)
-
-            let result = await callVariant(
-                inputParamsDict,
-                variantData[idx].inputParams!,
-                variantData[idx].optParams!,
-                appId || "",
-                variants[idx].baseId || "",
-                variantData[idx].isChatVariant
-                    ? testsetRowToChatMessages(evaluation.testset.csvdata[rowIndex], false)
-                    : [],
+        try {
+            setEvaluationStatus(EvaluationFlow.EVALUATION_STARTED)
+
+            const inputParamsDict = rows[rowIndex].inputs.reduce(
+                (acc: {[key: string]: any}, item) => {
+                    acc[item.input_name] = item.input_value
+                    return acc
+                },
+                {},
             )
-            if (variantData[idx].isChatVariant) result = contentToChatMessageString(result)
 
-            setRowValue(rowIndex, columnName as any, result)
-            await evaluate(rowIndex)
-            setShouldFetchResults(true)
-            if (rowIndex === rows.length - 1) {
-                message.success("Evaluation Results Saved")
+            const columnsDataNames = ["columnData0"]
+            let idx = 0
+
+            for (const columnName of columnsDataNames) {
+                setRowValue(rowIndex, "evaluationFlow", EvaluationFlow.COMPARISON_RUN_STARTED)
+
+                let result = await callVariant(
+                    inputParamsDict,
+                    variantData[idx].inputParams!,
+                    variantData[idx].optParams!,
+                    appId || "",
+                    variants[idx].baseId || "",
+                    variantData[idx].isChatVariant
+                        ? testsetRowToChatMessages(evaluation.testset.csvdata[rowIndex], false)
+                        : [],
+                )
+
+                if (variantData[idx].isChatVariant) {
+                    result = contentToChatMessageString(result)
+                }
+
+                setRowValue(rowIndex, columnName as any, result)
+                await evaluate(rowIndex)
+                setShouldFetchResults(true)
+
+                if (rowIndex === rows.length - 1) {
+                    message.success("Evaluation Results Saved")
+                }
+
+                idx++
             }
-            idx++
+
+            setEvaluationStatus(EvaluationFlow.EVALUATION_FINISHED)
```

Comment:
you have to place a condition here to only set this status when it's not being run form the run all button. Please check the code of SingleModelEvaluatoinTable to have an idea